ooniapi/services/oonirun/tests/test_oonirun.py

```diff
@@ -585,6 +606,251 @@ def test_oonirun_revisions(client, client_with_user_role):
+            "options": {
+                "HTTP3Enabled": True,
+            },
+            "is_background_run_enabled_default": False,
+            "is_manual_run_enabled_default": False,
+            "test_name": "web_connectivity",
+        },
+    ]
+    r = client_with_user_role.post("/api/v2/oonirun/links", json=z)
+    assert r.status_code == 422, r.json()
+
+    # Both targets and inputs_extra = error
+    z['nettests'] = [
+        {
+            "targets_name": "example_name",
+            "inputs_extra": [{}, {}],
+            "options": {
+                "HTTP3Enabled": True,
+            },
+            "is_background_run_enabled_default": False,
+            "is_manual_run_enabled_default": False,
+            "test_name": "web_connectivity",
+        },
+    ]
+    r = client_with_user_role.post("/api/v2/oonirun/links", json=z)
+    assert r.status_code == 422, r.json()
+
+def test_creation_with_targets_name(client_with_user_role):
+    z = deepcopy(SAMPLE_OONIRUN)
+    z['name'] = "Testing dynamic test lists calculation"
+    z['nettests'][0]['inputs'] = None
+    z['nettests'][0]['targets_name'] = "websites_list_prioritized"
+    z['nettests'] = z['nettests'][:1]
+
+    # Create
+    r = client_with_user_role.post("/api/v2/oonirun/links", json=z)
+    assert r.status_code == 200, r.json()
+    j = r.json()
+
+    # Retrieve
+    r = client_with_user_role.get(f"/api/v2/oonirun/links/{j['oonirun_link_id']}")
+    assert r.status_code == 200, r.json()
+    j = r.json()
+
+    # Does it have the targets name?
+    assert j['nettests'][0]['targets_name'] == "websites_list_prioritized", "Missing targets_name"
+
+    # now test that you can edit
+    z['nettests'][0]['targets_name'] = "new_value"
+    r = client_with_user_role.put(f"/api/v2/oonirun/links/{j['oonirun_link_id']}", json=z)
+    assert r.status_code == 200, r.json()
+
+    # Retrieve again
+    r = client_with_user_role.get(f"/api/v2/oonirun/links/{j['oonirun_link_id']}")
+    assert r.status_code == 200, r.json()
+    j = r.json()
+
+    assert j['nettests'][0]['targets_name'] == 'new_value', "Value of nettest should be changed by now"
+
+def test_dynamic_test_lists_calculation(client_with_user_role):
+    z = deepcopy(SAMPLE_OONIRUN)
+    z['name'] = "Testing dynamic test lists calculation"
+    z['nettests'][0]['inputs'] = None
+    z['nettests'][0]['targets_name'] = "websites_list_prioritized"
+    z['nettests'] = z['nettests'][:1]
+
+    r = client_with_user_role.post("/api/v2/oonirun/links", json=z)
+    assert r.status_code == 200, r.json()
+    j = r.json()
+
+    r = client_with_user_role.post(f"/api/v2/oonirun/links/{j['oonirun_link_id']}/engine-descriptor/latest", json=SAMPLE_META)
+    assert r.status_code == 200, r.json()
+
+    j = r.json()
+    assert j['nettests'][0]['targets_name'] == "websites_list_prioritized" 
+
+    # TODO(luis) Finish this test
+
+
+# TODO(luis) finish this test for checking the parsing of user agent headers 
```

Comment:
Are you planning to address these TODO in this PR or as following ones? If the later, then it would be a good idea to make an issue for them.

Reply:
I think we can mark it as complete, because I put that before doing integration tests and the simpler header format so I don't think there's something else to add in this tests